### ISSUE TITLE
Not able to compile Cockpit in container debian:2-bullseye for ARM64

### ISSUE THREAD

**Issue by ronaldoafonso** [cockpit-project/cockpit#15555]
Cockpit version: I'm using git tag 237 (git clone --branch 237 --depth 1 https://github.com/cockpit-project/cockpit.git)
OS: A Debian container, "debian:bullseye-20200908-slim" for ARM64 architecture.

I'm trying to compile the Cockpit Project and getting the following error:

CCLD     test-ssh-add
  CC       src/ws/remotectl-cockpitcertificate.o
  CC       src/ws/remotectl-remotectl-certificate.o
  CC       src/ws/remotectl-remotectl.o
  CCLD     remotectl
  GEN      cockpit-desktop
  WEBPACK  base1
  WEBPACK  apps
  WEBPACK  kdump
  WEBPACK  machines
make[1]: *** [Makefile:9850: dist/machines/stamp] Error 124
make[1]: Leaving directory '/cockpit'
make: *** [Makefile:3217: all] Error 2
The command '/bin/sh -c npm config set strict-ssl false &&     ./autogen.sh --prefix=/build &&     make &&     make install' returned a non-zero code: 2

I just pull the debian image and execute the following in a running container:

docker container run --platform "linux/arm64" -it debian:bullseye-20200908-slim bash
git clone --branch 237 --depth 1 https://github.com/cockpit-project/cockpit.git
cd ./cockpit
npm config set strict-ssl false
./autogen.sh --prefix=/build
make

After the executing of make ... I get the error.

**Comment by marusak** [comment]
I've tried just rebuilding tag 237 and it works in general.
I don't have arm64 machine though.
I wanted to try to use your reproducer (with just omitting ` --platform "linux/arm64"`) but the reproducer misses way to many steps. The container does not have `git`, `npm`, `autoreconf`... Can you please provide step by step reproducer of what you did? (I can install missing packages, but it might be that you are missing something and for that full reproducer would be useful).

**Comment by ronaldoafonso** [comment]
Hi,

I'm using a Dockerfile like this:

----------
ARG IMAGE_ARCH
ARG IMAGE_TAG_DEBIAN

FROM --platform=${IMAGE_ARCH} torizon/debian:${IMAGE_TAG_DEBIAN} AS builder

RUN apt-get update && \
    apt-get install -o Acquire::Check-Valid-Until=false \
                    -o Acquire::http::Dl-Limit=50 \
                    -o Acquire::https::Dl-Limit=50 \
                    -o Acquire::Retries=5 \
                    -y --no-install-recommends \
                    git \
                    npm \
                    autoconf \
                    automake \
                    libglib2.0-dev \
                    libsystemd-dev \
                    libjson-glib-dev \
                    gnutls-dev \
                    libkrb5-dev \
                    libssh-dev \
                    libpam0g-dev \
                    libpcp3-dev \
                    libpcp-import1-dev \
                    libpcp-pmda3-dev \
                    systemd \
                    gettext \
                    libxslt1-dev \
                    xsltproc \
                    xmlto \
                    sassc \
                    procps \
                    autopoint \
                    bison \
                    bzip2 \
                    docbook-xml \
                    docbook-xsl \
                    flex \
                    libaudit-dev \
                    libdb-dev \
                    libfl-dev \
                    libselinux1-dev \
                    libtool \
                    libxml2-utils \
                    make \
                    pkg-config \
                    sed \
                    w3m \
                    xz-utils \
                    sosreport

RUN git clone --branch 237 --depth 1 https://github.com/cockpit-project/cockpit.git

WORKDIR /cockpit

RUN npm config set strict-ssl false && \
    ./autogen.sh --prefix=/build --enable-debug && \
    make && \
    make install

RUN PACKAGES="apps kdump machines packagekit playground selinux" && \
    for PACKAGE in $PACKAGES; do \
        rm -rf /build/share/cockpit/$PACKAGE; \
    done

FROM --platform=${IMAGE_ARCH} torizon/debian:${IMAGE_TAG_DEBIAN}

RUN apt-get update && \
    apt-get install -o Acquire::Check-Valid-Until=false \
                    -o Acquire::http::Dl-Limit=50 \
                    -o Acquire::https::Dl-Limit=50 \
                    -o Acquire::Retries=5 \
                    -y --no-install-recommends \
                    libglib2.0-0 \
                    libjson-glib-1.0-0 \
                    libpcp3 \
                    libssh-4 \
                    systemd \
                    openssh-client \
                    sosreport \
                    procps \
                    inotify-tools

COPY --from=builder /build ./build/

RUN cp -dpRv /build/bin/* /bin && \
    cp -dpRv /build/etc/* /etc && \
    cp -dpRv /build/lib/* /lib && \
    cp -dpRv /build/var/* /var && \
    cp -dpRv /build/sbin/* /sbin

ADD cockpit.sh inotify.sh /sbin/

EXPOSE 9090

CMD ["/sbin/cockpit.sh"]

----------
And basically I built the image using the following docker-compose command:

`docker-compose build --build-arg IMAGE_ARCH="linux/arm64" --build-arg IMAGE_TAG_DEBIAN="bullseye-20200908-slim"
CMD ["/sbin/cockpit.sh"]`

Another thing, or question ... As you can see, I'm trying to remove some packages after building the image using the

`rm -rf /build/share/cockpit/$PACKAGE`

Is there a better way for doing it? I don't need these packages in my final image.

Thanks in advance for your help.

**Comment by ronaldoafonso** [comment]
Hi,

I've just found the problem. The thing was that the machine I was compiling Cockpit was taking so long to compile it. There are some "timeout" commands in the "Makefile.am" that simply times out if an operation takes so much time ... In my case more than 15 minutes ... and that was what was failing my Cockpit compilation. It was a little bit difficult to see it using just "make" as the command line. So I run "make V=1" and I was able to have a clue about the issue. I just increased the timeout time and it compiled successfully.

Thanks for the help and time anyway.
